### PR TITLE
doc: nrf: drivers: Add documentation for TX power ceiling

### PR DESCRIPTION
**Pull request by prsi98**
Add documentation for TX power ceiling options.

### CODE REVIEW

**Review by bama-nordic — APPROVED** [review]

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst [hunk 1]
```diff
@@ -76,9 +76,66 @@ Configuration
...
+
+The following parameters (8-bit unsigned values) are defined in the DTS overlay board files
+
+.. list-table:: DTS file parameters
+   :header-rows: 1
+
+   * - DTS parameter
+     - Description
+   * - max-pwr-2g-dsss
+     - Transmit power ceiling for DSSS data rate in 0.25dBm steps. This is applicable for all DSSS data rates.
```
Comment (richabp) [inline doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:121]: Please write the second sentence in the new line for all such instances. Also, add space between the numerical value and unit symbol.
```suggestion
     - Transmit power ceiling for DSSS data rate in 0.25 dBm steps.
       This is applicable for all DSSS data rates.
```

**Review by richabp — CHANGES_REQUESTED** [review]
Some style issues.

**Review by srkanordic — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst (hunk 1 above)
Comment (prsi98) [reply doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:121]: Made suggested changes.

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst [hunk 2]
```diff
@@ -76,9 +76,66 @@ Configuration
...
+			max-pwr-5g-mid-mcs0 = <0x38>;
+			max-pwr-5g-mid-mcs7 = <0x38>;
+			max-pwr-5g-high-mcs0 = <0x38>;
+			max-pwr-5g-high-mcs7 = <0x38>;
+		};
+
+	};
+
+
+The following parameters (8-bit unsigned values) are defined in the DTS overlay board files
```
[the remark this answers is not in the document]
Comment (prsi98) [reply doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:113]: Taken care.

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst [hunk 3]
```diff
@@ -76,9 +76,66 @@ Configuration
...
+
 .. options-from-kconfig::
    :show-type:
 
+DTS based
+^^^^^^^^^
+
+Maximum transmit power achieved on a nRF70 device based product depends on frequency band and operating channel. This varies from chip to chip as well as over different PCB designs. Multiple calibrations and checks are implemented to get consistency across channels and devices. However, these values have dependency on PCB design which may result in EVM (Error Vector Magnitude) and Spectral mask failures. To avoid this problem, a provision is made in software for user to specify the power ceiling at which EVM and Spectral mask is met for a given PCB design and are build-time parameters made available to driver through DTS overlay file.
+
+Below is an example of the DTS overlay file
```
[the remark this answers is not in the document]
Comment (prsi98) [reply doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:90]: Amended as per suggestion.

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst [hunk 4]
```diff
@@ -76,9 +76,66 @@ Configuration
...
+KConfig based
+^^^^^^^^^^^^^
+
 .. options-from-kconfig::
    :show-type:
 
+DTS based
+^^^^^^^^^
+
+Maximum transmit power achieved on a nRF70 device based product depends on frequency band and operating channel. This varies from chip to chip as well as over different PCB designs. Multiple calibrations and checks are implemented to get consistency across channels and devices. However, these values have dependency on PCB design which may result in EVM (Error Vector Magnitude) and Spectral mask failures. To avoid this problem, a provision is made in software for user to specify the power ceiling at which EVM and Spectral mask is met for a given PCB design and are build-time parameters made available to driver through DTS overlay file.
```
[the remark this answers is not in the document]
Comment (prsi98) [reply doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:88]: Incorporated changes.

File: doc/nrf/drivers/wifi/nrf700x/nrf700x.rst [hunk 5]
```diff
@@ -76,9 +76,66 @@ Configuration
...
 The nRF70 Series Wi-Fi driver has the following configuration options:
 
+KConfig based
+^^^^^^^^^^^^^
+
 .. options-from-kconfig::
    :show-type:
 
+DTS based
+^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (prsi98) [reply doc/nrf/drivers/wifi/nrf700x/nrf700x.rst:86]: Added requested changes.

**Review by richabp — APPROVED** [review]

**Pull request merged**